### PR TITLE
feat: extra withdrawals alerts

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts [hunk 1]
```diff
@@ -36,6 +37,14 @@ let lastTokenRebaseTimestamp = 0;
 let amountOfRequestedStETHSinceLastTokenRebase = new BigNumber(0);
 let lastBigRequestAfterRebaseAlertTimestamp = 0;
 
+let lastQueueOnParStakeLimitAlertTimestamp = 0;
```
Comment (dgusakov) [inline ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts:40]: What does "Par" in the naming means?

Comment (vgorkavenko) [reply ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts:40]: > What does "Par" in the naming means?

'on the same level', I guess. This term is taken from hackmd source

File: ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts [hunk 2]
```diff
@@ -195,6 +288,77 @@ async function handleUnfinalizedRequestNumber(
...
+
+  const withdrawalNFT = new ethers.Contract(
+    WITHDRAWAL_QUEUE_ADDRESS,
+    WITHDRAWAL_QUEUE_ABI,
+    ethersProvider
+  );
+
+  const unclaimedRequestsStatuses = (
+    await withdrawalNFT.functions.getWithdrawalStatus(
+      Array.from(unclaimedRequests.values()),
```
Comment (dgusakov) [inline ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts:311]: Can't find handler that will update unclaimedRequests to include new requests

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts [hunk 3]
```diff
@@ -195,6 +288,77 @@ async function handleUnfinalizedRequestNumber(
...
+      if (!unclaimedRequests.has(reqId)) {
+        unclaimedRequests.add(reqId);
+        unclaimedSize = unclaimedSize.plus(
+          new BigNumber(String(reqStatus.amountOfStETH))
+        );
+      }
+    }
+  }
+
+  const totalFinalizedSize = claimedSize.plus(unclaimedSize);
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts:356]: fixed

File: ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts (hunk 2 above)
Comment (vgorkavenko) [reply ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts]: fixed

File: ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts [hunk 4]
```diff
@@ -195,6 +294,87 @@ async function handleUnfinalizedRequestNumber(
...
+
+  const unclaimedReqIds: number[] = [];
+  const outdatedReqIds: number[] = [];
+  finalizedWithdrawalRequests.forEach((req, id) => {
+    const isOutdated =
+      now - Number(req.timestamp) > UNCLAIMED_REQUESTS_TIME_WINDOW;
+    if (isOutdated) {
+      outdatedReqIds.push(id);
+    }
+    if (!isOutdated && !req.claimed) {
```
Comment (dgusakov) [inline ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts:323]: We should keep ALL unclaimed requests and look at the ALL unclaimed to claimed over the last 2 weeks requests

Comment (vgorkavenko) [reply ethereum-operations/src/subagents/withdrawals/agent-withdrawals.ts]: fixed

**Review by dgusakov — APPROVED** [review]

**Pull request merged**